lib/Dialect/AIEVec/Transforms/VectorToAIEVecConversions.cpp

```diff
@@ -46,6 +46,49 @@ using namespace xilinx::aievec;
 // Utility functions
 //===----------------------------------------------------------------------===//
 
+static bool isNarrowingOp(Operation *op) {
+  if (isa<arith::TruncFOp>(op) || isa<arith::TruncIOp>(op))
+    return true;
+
+  if (auto srsOp = dyn_cast<aievec::SRSOp>(op)) {
+    auto srsOpSrcOp = srsOp.getSource().getDefiningOp();
+    if (isa<aievec::UPSOp>(srsOpSrcOp) || isa<aievec::CastOp>(srsOpSrcOp))
```

Comment:
Suppose that we have UPS from int8 => int32 followed by SRS from int32 => int16.
Would it be OK to return true in this case?

Reply:
I don't think we should because that's not a narrowing operation (int8 -> int16), but in this particular case it's looking for an `SRS` followed by either a `UPS` or a `aievec.cast`, which are the two possible conversion for `trunc` ops, depending on the operand types.